markdown_lab/core/errors.py

```
@@ -0,0 +1,327 @@
+        f"Memory usage ({current:,} bytes) exceeds limit ({limit:,} bytes)",
+        resource_type="memory",
+        current_usage=current,
+        limit=limit,
+        error_code="MEMORY_LIMIT_EXCEEDED"
+    )
+
+
+# Error handling utilities
+def handle_request_exception(exception: Exception, url: str, retry_count: int = 0) -> NetworkError:
+    """Convert various request exceptions to standardized NetworkError."""
+    import requests
```

Pylint (3.3.7)

[convention] Import outside toplevel (requests)

(C0415)